### PR TITLE
Change TS Target to ES2015 and Enable Strict Mode

### PR DESCRIPTION
**Pull request by kuasha420**
Hello, 

Thanks again for this amazing template. 

One thing I consistently do after starting a new project is make this two changes for the following reason and was wondering if you'd consider adding them to the template. 

## Rationale 

1. nextjs itself only supports modern browser with es6+ support and IE11 with polyfills. We can safely use es2015 as target as compilation is done by nextjs/babel. This allows us to use advanced ecmascript features like **using spread operator to convert an Iterator to an Array without error or extra compiler flags.**

2. enabling strict mode helps us write better and safer code and it's recommended in most cases.

Let me know if any further changes needs to be done for this to be merged. 

## Changes Made

1. changed target to es2015 (es6)
2. changed strict mode to true
3. added type to src/pages/_app.tsx to pass strict mode (noImplicitAny)

## Test Plan

- [x] pass type check `yarn type-check`
- [x] start dev server `yarn dev`
- [x] prod build `yarn build`

### CODE REVIEW

File: src/pages/_app.tsx [hunk 1]
```diff
@@ -1,5 +1,6 @@
 import '@/styles/global.css';
+import { AppProps } from 'next/app';
```
Comment (jpedroschmitz) [inline src/pages/_app.tsx:2]: I know it's a small thing, but can you put this import on the top of the file? 😅

**Review by jpedroschmitz — CHANGES_REQUESTED** [review]
Hey @kuasha420, thanks for the PR. I think it's ok to enable strict mode for TypeScript. If anyone wants to disable it, can change it later.

About the target option, I don't see any reason not to change it since Next.js (through Babel) will compile it.

I requested just a small change on the `_app.tsx` file.

File: src/pages/_app.tsx (hunk 1 above)
Comment (kuasha420) [reply src/pages/_app.tsx:1]: I've fixed it. 

Just a heads up, VSCode's organize import will put the absolute imports '@/' at the top before library imports for some reason.

I haven't found any good way of fixing that other than changing absolute import prefix from @ to ~.

**Review by jpedroschmitz — APPROVED** [review]
Thanks, @kuasha420 💜

**Pull request merged**